### ISSUE TITLE
dune install: ocamlfind point to the wrong directory in a local switch

### ISSUE THREAD

**Issue by FardaleM** [ocaml/dune#3025]
## Expected Behavior
Install file at the correct place.

## Actual Behavior
In a local switch in which ocamlfind is not installed, dune install use ocamlfind from the current global switch to choose where to install file and try to install package in the global switch instead of the local one.

## Reproduction
1. Create a global switch and install ocamlfind in it.
1. Create a local switch and try to install lablqml.

## Specifications

- Version of `dune` (output of `dune --version`): 2.1.2
- Version of `ocaml` (output of `ocamlc --version`): 4.09.0
- Operating system (distribution and version): Debian sid

**Comment by diml** [comment]
Indeed, that's a known issue. Currently, when you have such "layered" installation, dune doesn't differentiate between the various installation. And since historically `ocamlfind` has been the authoritative tool for installing OCaml libraries, that's what `dune` prefers to use when it is available.

We could do some work on dune to make it detect such layered installation and support them cleanly. I have a few ideas in mind but they'd need a bit more thoughts.

In the meantime, the easiest workaround is to install `ocamlfind` in the local switch.
